crates/ir/src/register/chill.rs

```diff
@@ -0,0 +1,19 @@
+/// Container which holds the registers which are most optimal to chill in the current context.
+/// # Remarks
+/// This is a hint to the JIT compiler to chill one of these registers if all of them are busy, by previous
+/// instructions.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct RegisterChill {
+    _default: (),
+}
```

Comment:
I can't find any definition on what chilling registers means. We talked about it offline and therefore I more or less know the definition, but it would be really helpful if you could add a module-wide comment here that would define what register chilling is (and link to this module from doc comments which reference chilling.) Like so (example taken from another part of the docs):

```rs
/// - [`RegisterChill`] - most optimal registers to [chill][crate::register::chill]
```

Reply:
Also, I believe the field `_default` should probably be renamed to `_private` or something like that, because from what I could tell the only reason to have it is to prevent construction by modules other than `rokugo_ir::register::chill`.